src/Stack/Storage.hs

```diff
@@ -114,6 +123,33 @@ DockerImageExeCache
   compatible Bool
   DockerImageExeCacheUnique imageHash exePath exeTimestamp
   deriving Show
+
+CompilerCache
+  actualVersion ActualCompiler
+  arch Text
+
+  -- Include ghc executable size and modified time for sanity checking entries
+  ghcPath FilePath
+  ghcSize Int64
+  ghcModified Int64
+
+  ghcPkgPath FilePath
+  runghcPath FilePath
+  haddockPath FilePath
+
+  cabalVersion Text
+  globalDb FilePath
+  globalDbCacheSize Int64
+  globalDbCacheModified Int64
+  info ByteString
+
+  -- This is the ugliest part of this table, simply storing a Show/Read version of the
+  -- data. We could do a better job with normalized data and proper table structure.
+  -- However, recomputing this value in the future if the data representation changes
+  -- is very cheap, so we'll take the easy way out for now.
+  globalDump Text
+
+  UniqueGhcInfo ghcPath
```

Comment:
Maybe call this `UniqueCompilerInfo` to match the table name?

Reply:
Oh, good point, I'll update and then merge.